internal/handler/wallet/precondition_test.go

```diff
@@ -0,0 +1,104 @@
+package wallet
+
+import (
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/go-openapi/loads"
+	"github.com/massalabs/thyra-plugin-massa-wallet/api/server/restapi"
+	"github.com/massalabs/thyra-plugin-massa-wallet/api/server/restapi/operations"
+)
+
+// configureAPIServerPrecondition sets up the API server with the required endpoints
+// and handler functions for the precondition tests
+func configureAPIServerPrecondition() (*operations.MassaWalletAPI, error) {
+	// Load the Swagger specification
+	swaggerSpec, err := loads.Analyzed(restapi.SwaggerJSON, "")
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a new MassaWalletAPI instance
+	localAPI := operations.NewMassaWalletAPI(swaggerSpec)
+	server := restapi.NewServer(localAPI)
+
+	// Create a sync.Map to store the wallets
+	var walletStorage sync.Map
+
+	// Set the handler function for the create wallet endpoint
+	localAPI.RestWalletCreateHandler = NewCreate(&walletStorage)
+
+	// Configure the API endpoints
+	server.ConfigureAPI()
+
+	return localAPI, nil
+}
+
+// createTestWallet creates a new wallet which will be used as a preconditon to run other tests
+func createTestWallet(t *testing.T) {
```

Comment:
You should diretcly use the function that test the wallet creation or write by hand a JSON file